### ISSUE TITLE
How to make RankingLabelSizeTransformer work?

### ISSUE THREAD

**Issue by Kalethars** [gephi/gephi#1606]
```
    Column nodeSizeCol = graphModel.getNodeTable().getColumn("Weight");
    Function sizeRanking = appearanceModel.getNodeFunction(graph, nodeSizeCol, RankingNodeSizeTransformer.class);
    RankingNodeSizeTransformer sizeTransformer = (RankingNodeSizeTransformer) sizeRanking.getTransformer();
    sizeTransformer.setMaxSize(80);
    sizeTransformer.setMinSize(10);
    appearanceController.transform(sizeRanking);


    Column labelSizeCol = graphModel.getNodeTable().getColumn("Weight");
    Function sizeRanking2 = appearanceModel.getNodeFunction(graph, labelSizeCol, RankingLabelSizeTransformer.class);
    RankingLabelSizeTransformer sizeTransformer2 = (RankingLabelSizeTransformer) sizeRanking2.getTransformer();
    sizeTransformer2.setMaxSize(4);
    sizeTransformer2.setMinSize(1);
    appearanceController.transform(sizeRanking2);


    model.getProperties().putValue(PreviewProperty.SHOW_NODE_LABELS, Boolean.TRUE);
    model.getProperties().putValue(PreviewProperty.NODE_LABEL_FONT, new java.awt.Font("Comic Sans MS", 0, 12));
    model.getProperties().putValue(PreviewProperty.NODE_LABEL_PROPORTIONAL_SIZE, Boolean.FALSE);
```

I have written these codes, and I find the node sizes change while the label sizes are 12. I have tried this in Gephi GUI and it works well. How can I make label size adjustment work?

**Comment by eduramiba** [comment]
I think NODE_LABEL_PROPORTIONAL_SIZE should be true instead of false.

**Comment by Kalethars** [comment]
@eduramiba 
I set this value False in GUI, and it works.
After I set that True in Java Code, the label sizes only adjust by default proportion.

**Comment by eduramiba** [comment]
Oh I misunderstood. Did you refresh the preview? You should do it after applying all transformers and properties, like in the GUI.

**Comment by Kalethars** [comment]
@eduramiba 
I refresh the preview.

**Comment by eduramiba** [comment]
I think you now need to call refresh on `previewModel` too.

**Comment by Kalethars** [comment]
@eduramiba 
how to call refresh on previewModel?
I just write that at the end of the program.

        model.getProperties().putValue(PreviewProperty.SHOW_NODE_LABELS, Boolean.TRUE);
        model.getProperties().putValue(PreviewProperty.NODE_LABEL_FONT, new java.awt.Font("Comic Sans MS", 0, 12));
        model.getProperties().putValue(PreviewProperty.NODE_LABEL_PROPORTIONAL_SIZE, Boolean.FALSE);
        model.getProperties().putValue(PreviewProperty.EDGE_RESCALE_WEIGHT, Boolean.TRUE);
        model.getProperties().putValue(PreviewProperty.EDGE_THICKNESS, new Float(5f));
        model.getProperties().putValue(PreviewProperty.NODE_BORDER_WIDTH, new Float(0f));
        model.getProperties().putValue(PreviewProperty.EDGE_CURVED, Boolean.FALSE);
        model.getProperties().putValue(PreviewProperty.NODE_LABEL_OUTLINE_SIZE, new Float(0f));
        model.getProperties().putValue(PreviewProperty.NODE_LABEL_OUTLINE_OPACITY, new Float(100f));
        //model.getProperties().putValue(PreviewProperty.NODE_LABEL_OUTLINE_COLOR, new DependantColor(new Color(0x000000)));
        //model.getProperties().putValue(PreviewProperty.NODE_LABEL_COLOR, new DependantOriginalColor(DependantOriginalColor.Mode.PARENT));
        model.getProperties().putValue(PreviewProperty.NODE_LABEL_COLOR, new DependantOriginalColor(new Color(0x000000)));
        previewController.refreshPreview();

        G2DTarget target = (G2DTarget) previewController.getRenderTarget(RenderTarget.G2D_TARGET);
        PreviewSketch previewSketch = new PreviewSketch(target);
        previewController.refreshPreview();
        previewSketch.resetZoom();
        
        ExportController ec = Lookup.getDefault().lookup(ExportController.class);
        File svgFile = new File("src\\affiliations\\"+authorID+".svg");

        ec.exportFile(svgFile);

**Comment by eduramiba** [comment]
Sorry that was wrong, your example seems fine. Maybe there is a bug that makes it necessary to go to overview for labels to apply size.
Does it work in GUI if you open it in preview, apply ranking in preview too (open Appearance window from Window menu) and refresh without going to overview?

**Comment by Kalethars** [comment]
@eduramiba 
It works well in GUI. I always adjust parameters in GUI before writing code in Java.

**Comment by eduramiba** [comment]
Yeah but without going to overview at all?

**Comment by Kalethars** [comment]
@eduramiba 
I adjust other parameters such as node colors in appearance window in overview.

**Comment by eduramiba** [comment]
@Kalethars That's what it might be working in the GUI and not in code. So it might fail in the GUI if you never open Overview. In order to do that, use appearance window from preview, opening it in the Window menu.

**Comment by Kalethars** [comment]
@eduramiba 
Wow, both label size and label color fail.

**Comment by eduramiba** [comment]
There must be a overview dependent bug then.

**Comment by Kalethars** [comment]
@eduramiba 
Gonna fix it in 0.9.2 release?

**Comment by eduramiba** [comment]
We will have to investigate the problem and see if a fix is doable.

**Comment by Kalethars** [comment]
@eduramiba 
Thank you anyway :-) At least I know this is not my coding error.

**Comment by Kalethars** [comment]
@eduramiba 
So can I open overview in api just like what I do in GUI? Maybe that will make me able to adjust label size and color.

**Comment by eduramiba** [comment]
Actually the overview visualization module is not very modular and cannot be included in the toolkit for the moment. This kind of issues with overview are known, and it's a hard wishlist task to improve that.

So a first step will be to determine what's the problem with labels and fix it separately.

**Comment by eduramiba** [comment]
Fixed for 0.9.2, please try with the latest toolkit snapshot tomorrow, after the nightly build is done.
